### PR TITLE
Migrate slider component to chakra UI

### PR DESCRIPTION
**Pull request by lawlesx**
## Description
Migrated `Slider.tsx` component to Chakra UI

### Screenshots
**Dark Mode**
Before
Screenshot 2022-10-16 at 12 02 45 AM
After
Screenshot 2022-10-16 at 12 02 34 AM

**Light Mode**
Before
Screenshot 2022-10-16 at 12 03 01 AM
After
Screenshot 2022-10-16 at 12 02 55 AM

## Related Issue

 Resolves Issue #8051

### CODE REVIEW

File: src/components/Slider.tsx [hunk 1]
```diff
@@ -146,42 +52,89 @@ const Slider: React.FC<IProps> = ({ children, onSlideChange }) => {
...
+      borderWidth="1px"
+      borderStyle="solid"
+      borderRadius="0.3rem"
+      w="full"
+      bg="sliderBg"
+    >
+      <Box overflow="hidden" ref={emblaRef}>
+        <Flex>{children}</Flex>
+      </Box>
+      <Flex
```
Comment (pettinarip) [inline src/components/Slider.tsx:67]: I think we should position this as `absolute` as before. If not it looks weird.

image

The dots should be centered with the buttons. Take a look at [the prev implementation](https://github.com/ethereum/ethereum-org-website/blob/dev/src/components/Slider.tsx#L52).

File: src/components/Slider.tsx [hunk 2]
```diff
@@ -146,42 +52,89 @@ const Slider: React.FC<IProps> = ({ children, onSlideChange }) => {
...
+        justifyContent={{ sm: "left", base: "center" }}
+        mb={{ sm: "0", base: "1rem" }}
```
Comment (pettinarip) [inline src/components/Slider.tsx:69]:
```suggestion
        mb={{ sm: 0, base: 4 }}
```

File: src/components/Slider.tsx [hunk 3]
```diff
@@ -146,42 +52,89 @@ const Slider: React.FC<IProps> = ({ children, onSlideChange }) => {
...
-                ? theme.colors.slider.btnColor
-                : theme.colors.slider.btnColorDisabled
-            }
-          />
-        </Button>
-      </Buttons>
-      <Dots>
+    <Box
+      position="relative"
+      p="2rem"
```
Comment (pettinarip) [inline src/components/Slider.tsx:57]:
```suggestion
      p={8}
```

File: src/components/Slider.tsx [hunk 4]
```diff
@@ -146,42 +52,89 @@ const Slider: React.FC<IProps> = ({ children, onSlideChange }) => {
...
+      borderWidth="1px"
+      borderStyle="solid"
+      borderRadius="0.3rem"
```
Comment (pettinarip) [inline src/components/Slider.tsx:60]: Lets use `base` here (which is `0.25rem`) to be consistent with the rest of the site.

```suggestion
      borderRadius="base"
```

File: src/theme.ts [hunk 5]
```diff
@@ -369,16 +367,14 @@ const darkColors = {
...
-    btnColorDisabled: "#737373",
-  },
+  sliderBg: "#191919",
+  sliderBorder: "#404040",
+  sliderDot: "#A4A4A4",
+  sliderDotActive: "#FF7324",
+  sliderBtnBg: "#404040",
+  sliderBtnColor: white,
+  sliderBtnBgDisabled: "#404040",
+  sliderBtnColorDisabled: "#737373",
```
Comment (pettinarip) [inline src/theme.ts:377]: Why do we need these changes? if not used please revert

**Review by pettinarip — CHANGES_REQUESTED** [review]
Good job @lawlesx. The dots look a bit misaligned. I've left a possible solution.

File: src/theme.ts (hunk 5 above)
Comment (lawlesx) [reply src/theme.ts:377]: Screenshot 2022-10-22 at 5 01 44 PM

In the previous implementation I was not able to apply the colors
Example:
```ts
bg="slider.bg"
```
This change was not reflected

I tried to reach out in [discord](https://discord.com/channels/714888181740339261/870313128242602004/1028259754360983562) but got no solution so the did the above change

File: src/components/Slider.tsx (hunk 1 above)
Comment (lawlesx) [reply src/components/Slider.tsx:67]: Sure the buttons looks odd and I did try the prev implementation but there was some issues with the alignment so I made it in a presentable way.

Also as you said the button should be aligned center.
I pushed some changes that should fix it

## Attaching screen shots for new and old changes
### New Pushed changes as this comment date
Screenshot 2022-10-22 at 6 26 32 PM

### Old changes as in the live [website](https://ethereum.org/en/what-is-ethereum/)
Screenshot 2022-10-22 at 5 13 01 PM

Comment (lawlesx) [reply src/components/Slider.tsx:67]: Also I made sure it doesn't break for mobile view

**Review by pettinarip — CHANGES_REQUESTED** [review]
Thanks for making the adjustments @lawlesx.

The last thing I see is that button's size is not the previous one. They should be `35px` instead of `40px`.

File: src/components/Slider.tsx [hunk 6]
```diff
@@ -146,42 +52,93 @@ const Slider: React.FC<IProps> = ({ children, onSlideChange }) => {
...
+          aria-label="MdChevronRight"
+          onClick={scrollNext}
+          icon={<MdChevronRight fontSize={24} focusable={true} />}
+          isRound
+          _hover={{ boxShadow: "none" }}
+          _focus={{ boxShadow: "none" }}
+          bg={nextBtnEnabled ? "sliderBtnBg" : "sliderBtnBgDisabled"}
+          height="40px"
+          color={nextBtnEnabled ? "sliderBtnColor" : "sliderBtnColorDisabled"}
+        />
```
[the remark this answers is not in the document]
Comment (lawlesx) [reply src/components/Slider.tsx:93]: While appling `height` and `width` to `35px` the the buttons kind of breaks.
Screenshot 2022-10-25 at 9 18 23 PM

So I fixed it by applying `size="sm"`
Screenshot 2022-10-25 at 9 20 51 PM

I feel it looks close to the previous button

[the remark this answers is not in the document]
Comment (pettinarip) [reply src/components/Slider.tsx:93]: Nice! I like it even more (using standard Chakra values). Gj

**Review by pettinarip — APPROVED** [review]
Awesome @lawlesx this looks great.

**Pull request merged**